plugin/vcs/vcs.go

```diff
@@ -147,18 +162,22 @@ type Provider interface {
 	//
 	// Similar to CreateFile except it overwrites an existing file. The fileCommit shoud includes the "LastCommitID" field which is used to detect conflicting writes.
 	OverwriteFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath string, fileCommit FileCommitCreate) error
-	// Reads the file content. Returns an io.ReadCloser on success. If file does not exist, returns NotFound error.
+	// Reads the file data
 	//
-	// oauthCtx: OAuth context to read the file content
+	// oauthCtx: OAuth context to fetch the file data
 	// instanceURL: VCS instance URL
 	// repositoryID: the repository ID from the external VCS system (note this is NOT the ID of Bytebase's own repository resource)
 	// filePath: file path to be read
-	// commitID: the specific version to be read
-	ReadFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, commitID string) (string, error)
-	// Reads the file metadata. Returns the file meta on success.
+	// ref: the name of branch, tag or commit
+	ReadFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, ref string) (*File, error)
```

Comment:
Based on the way it structures,

This one should be called ReadFileMeta and the type should be FileMeta, the FileMeta should not contain the content field.

The underlying implementation like GitLab might just use a single API and that's fine we have a single ReadFile inside the GitLab module.

But from VCS caller, it's better to set the boundary between read file meta and file content.

Reply:
Updated, PTAL.